packages/klighd-core/src/sidebar/sidebar.tsx

```diff
@@ -110,12 +118,28 @@ export class Sidebar extends AbstractUIExtension {
         // Show or hide the panel
         if (currentPanel) {
             this.containerElement.classList.add('sidebar--open')
+            // Set width of sidebar to maximum width of all opened panels.
+            this.maxWidth = Math.max(this.maxWidth, this.containerElement.clientWidth)
+            document.documentElement.style.setProperty('--sidebar-width', `${this.maxWidth}px`)
         } else {
             this.containerElement.classList.remove('sidebar--open')
+            // Reset sidebar width when closed
+            this.maxWidth = Math.max(this.maxWidth, this.containerElement.clientWidth)
+            document.documentElement.style.setProperty('--sidebar-width', '0px')
+        }
+        // Find the canvas element and dispatch an action to re-initialize the canvas bounds.
+        // This works since the previous css property will correctly set the width of the canvas.
+        const canvas = (content.elm as HTMLElement).parentElement?.parentElement
+        if (canvas) {
```

Comment:
To avoid potential errors and a call of the InitializeCanvasBoundsAction with invalid parameters, this should also check if canvas is of the correct type and contains clientLeft, etc.